### ISSUE TITLE
Option to Map SAML UID attribut to username

### ISSUE THREAD

**Issue by f00wl** [askvortsov1/flarum-saml#11]
Hey, first of all thanks for your work!

Our [IDP](https://forge.tedomum.net/acides/hiboo/hiboo) is able to provide multiple profile for a same user, this is very useful, but we have to ensure that on the application side username match the profile name define in the IDP.

In order to use this flarum extension to connect our flarum instance to our IDP, we need it to set the username automatically during the singup workflow.

Here is a naive solution that works for us, mapping the username to the standard UID Attribute Type `0.9.2342.19200300.100.1.1` provided by the IDP in the SAML response:

``` diff
index 9e96679..4482d11 100644
--- a/src/Controllers/ACSController.php
+++ b/src/Controllers/ACSController.php
@@ -97,6 +97,8 @@ class ACSController implements RequestHandlerInterface
             return new HtmlResponse('Email not provided.');
         }

+        $uid = $saml->getAttribute('urn:oid:0.9.2342.19200300.100.1.1')[0];
+
         $masquerade_attributes = [];
         foreach ($attributes as $key => $attribute) {
             if ($key != 'avatar' && $key != 'bio' && $key != 'groups') {
@@ -122,11 +124,15 @@ class ACSController implements RequestHandlerInterface
         return $this->response->make(
             'saml-sso',
             $saml->getNameId(),
-            function (Registration $registration) use ($avatar, $email) {
+            function (Registration $registration) use ($avatar, $email, $uid) {
                 $registration
                     ->provideTrustedEmail($email)
-                    ->suggestUsername('')
+                    ->suggestUsername($uid)
                     ->setPayload([]);
+                if ($uid != "") {
+                    $registration
+                        ->provide('username', $uid);
+                }

                 if ($avatar) {
                     $registration->provideAvatar($avatar);
```

![screen](https://user-images.githubusercontent.com/48769271/131258546-987ede85-1603-4360-9aef-da97edf79c1c.png)

It would be perfect to have such a functionality as an option in admin panel!

**Comment by askvortsov1** [comment]
Added in v2.3.1. I don't have time to set up a testing environment locally unfortunately so this is untested, so please let me know if you run into issues!

**Comment by Ornanovitch** [comment]
We'll let you know ASAP :)

**Comment by f00wl** [comment]
Thanks you for your fast reply, just a minor fix and it will be great!

**Comment by askvortsov1** [comment]
Published v2.4.0, thank you for the suggestion and contribution!
